### ISSUE TITLE
[Feature]: Support headers for reverse proxy authentication (Identity Aware Proxies)

### ISSUE THREAD

**Issue by raags** [appsmithorg/appsmith#9915]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Identity Aware Proxies (IAP) are used to implement the Zero Trust security model which uses the user identity instead of network-based as done by traditional VPNs.

These set specific headers, like X-USER-NAME, which is parsed by the backend to authenticate/create the user. For e.g. Gafana supports this: https://grafana.com/docs/grafana/latest/auth/auth-proxy/

### Why should this be worked on?

This feature would allow deploying appsmith in organizations that have implemented IAP solutions, where authentication happens in a single place.

**Comment by Nikhil-Nandagopal** [comment]
@raags thanks for the feature request. Is this a form of SSO?

**Comment by raags** [comment]
Not necessarily - you're essentially offloading the auth to a proxy, which adds the appropriate headers to allow the server to recognise and authorize (e.g. via group mappings) the user. This allows you to keep auth in one place (among other benefits), instead of configuring every app with separate keys. The app also doesn't need to support every SSO protocol.

**Comment by Nikhil-Nandagopal** [comment]
@raags sure but it does seem like a type of SSO where essentially all the appsmith authentication flows are now proxied through a 3rd party service and that service is what all the backend calls have to interface with to authenticate a user. Am I right in understanding that?

**Comment by raags** [comment]
Yes, I suppose so - but I suspect it'll be easier to implement (not familiar with appsmith code though)

**Comment by jpwilksch** [comment]
> @raags sure but it does seem like a type of SSO where essentially all the appsmith authentication flows are now proxied through a 3rd party service and **that service is what all the backend calls have to interface with to authenticate a user**. Am I right in understanding that?

If I understand @raags correctly, the part in bold is incorrect. It is authentication _to Appsmith_ that is proxied, not authentication with backend services with which Appsmith interacts.

I too would also like to see this added.

**Comment by GlennMatthys** [comment]
Related: https://github.com/appsmithorg/appsmith/issues/8329

**Comment by LauridsJMikkelsen** [comment]
Any news on this? Would also greatly appreciate this feature

**Comment by sebastiangeislerlipo** [comment]
is there any time to expect this?
